src/dejsonlz4.c

```diff
@@ -61,7 +62,19 @@ void *file_to_mem(const char *fname, size_t *out_size)
 {
     size_t fsize = 0;
     void *rv = 0;
-    FILE *f = fopen(fname, "rb");
+    FILE *f;
+    if (fname) {
+        f = fopen(fname, "rb");
+    } else {
+#ifdef _WIN32
+        /* Try changing stdout to binary mode, or else it messes with EOL chars.
+         * Might not matter much for Mozilla files since they're text.
+         */
+        if (_setmode(_fileno(stdin), O_BINARY) == -1)
+            fprintf(stderr, "Warning: cannot set input to binary mode\n");
+#endif
+        f = stdin;
```

Comment:
I have not tested it, but even if it works on some cases, as far as I know (do correct me if I'm wrong) stdin is not generally seekable, and reading its size using `ftell` would not work. Due to this, this function does not currently lend itself, IMO, to joint file and stdin reads.

What I think would be more correct here (again, do correct me if I'm wrong) is either modifying this function to read till `EOF` while reallocating and eventually making sure there were no errors along the way, or write a new function which only reads stdin till EOF to memory, and from `main` decide which function to call depending if you have or don't have `iname`.

Sounds reasonable?

Reply:
Oops, I did not look at the code thoroughly, since it strangely worked. Let me correct that ...